railties/lib/rails/generators/rails/app/templates/Dockerfile.tt

```diff
@@ -0,0 +1,50 @@
+# Make sure it matches the Ruby version in .ruby-version and Gemfile
+ARG RUBY_VERSION=<%= Gem.ruby_version %>
+FROM ruby:$RUBY_VERSION
+
+<% if using_node? -%>
+# Install JavaScript dependencies and libvips for Active Storage
+ARG NODE_MAJOR_VERSION=19
+RUN curl -sL https://deb.nodesource.com/setup_$NODE_MAJOR_VERSION.x | bash -
+RUN apt-get update -qq && \
+    apt-get install -y build-essential libvips nodejs && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/* /usr/share/doc /usr/share/man && \
+    npm install -g yarn
+
+<% elsif !skip_active_storage? -%>
+# Install libvips for Active Storage preview support
+RUN apt-get update -qq && \
+    apt-get install -y build-essential libvips && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/* /usr/share/doc /usr/share/man
+
+<% end -%>
+# Rails app lives here
+WORKDIR /rails
+
+# Set production environment
+ENV RAILS_LOG_TO_STDOUT="1" \
+    RAILS_SERVE_STATIC_FILES="true" \
+    RAILS_ENV="production" \
+    BUNDLE_WITHOUT="development:test"
+
+# Install application gems
+COPY Gemfile Gemfile.lock ./
+RUN bundle install
+
+# Copy application code
+COPY . .
+
+<% if depend_on_bootsnap? -%>
+# Precompile bootsnap code for faster boot times
+RUN bundle exec bootsnap precompile --gemfile app/ lib/
+
+<% end -%>
+# Precompiling assets for production without requiring secret RAILS_MASTER_KEY
+RUN SECRET_KEY_BASE_DUMMY=1 bundle exec rails assets:precompile
+
+# Entrypoint prepares database and starts app on 0.0.0.0:3000 by default,
+# but can also take a rails command, like "console" or "runner" to start instead.
+ENTRYPOINT ["/rails/bin/docker-entrypoint"]
```

Comment:
is it using the root user ?

Reply:
Yes. Mapping cleanly for non-root use-cases is an exercise in frustration especially for Linux where it needs to map to a host UID / GID in /etc/hosts.

If not running as root is important, run it via Podman. 

If this is intended for deployment, most hosting providers aren't giving you root level to the full server. Just within the container you're given so I don't think it's a huge deal.

Reply:
There are plenty of risks with rails running as root within a container, even is a syscap constrained environment (like the cloud). If a container is compromised, the attacker now has the ability to install packages, read and write arbitrary files within the container.

Reply:
@ianks thanks. This makes sense. I generally run all my docker containers as non-root users just because it's generally good practice, didn't realize there was real security concerns with it on hosting providers considering I never see dockerfiles from them as non-root users. Apologies for brushing this off and thank you for taking the time.